templatestore/frontend/src/components/TinyUrlComponent.js

```diff
@@ -0,0 +1,172 @@
+        visitedCopy[givenId] = 0;
+        props.updateItems(newItems);
+        props.updateVisited(visitedCopy);
+    }
+
+    function handleChange(e, id) {
+        e.persist();
+        if (e.target.value === "") {
+            props.showAlerts("Choose valid Url and expiry");
+            return;
+        }
+        let newItems = [...props.items];
+        let visitedCopy = { ...props.visited };
+        if (e.target.name === 'urlKey') {
+            visitedCopy[e.target.value] = 1;
+            if (id !== "") {
+                visitedCopy[id] = 0;
+            }
+            props.updateVisited(visitedCopy);
+        }
+        let index = newItems.findIndex(item => item.urlKey === id);
+        newItems[index][e.target.name] = e.target.value;
+        props.updateItems(newItems);
+    }
+
+    function arrayChanges() {
+        const isSame = (a, b) => a.urlKey === b.urlKey && a.expiry === b.expiry;
+        // Get props.items that only occur in the left array,
+        // using the compareFunction to determine equality.
+        const onlyInLeft = (left, right, compareFunction) =>
+            left.filter(leftValue =>
+                !right.some(rightValue =>
+                    compareFunction(leftValue, rightValue)));
+        let addedArray, removedArray;
+        if (tinyUrlObjLocal) {
+            removedArray = onlyInLeft(tinyUrlObjLocal, props.items, isSame);
+            addedArray = onlyInLeft(props.items, tinyUrlObjLocal, isSame);
+        } else {
+            removedArray = [];
+            addedArray = props.items;
+        }
+        return { itemsRemoved: removedArray,
+            itemsAdded: addedArray };
+    }
+
+    function valueSelected(e) {
+        e.preventDefault();
+        let data = {};
+        for (let i = 0; i < props.items.length; i++) {
+            if (props.items[i].urlKey === "" || props.items[i].expiry === "") {
+                props.showAlerts("Can't leave url or expiry blank");
+                return;
+            }
+        }
+        let changes = arrayChanges();
+        if (changes.itemsAdded.length === 0 && changes.itemsRemoved.length === 0) {
+            props.showAlerts("Please modify something");
+            return;
+        }
+
+        for (let i = 0; i < changes.itemsRemoved.length; i++) {
+            const index = tinyUrlObjLocal.findIndex(obj => obj.urlKey === changes.itemsRemoved[i].urlKey); // eslint-disable-line
+            if (index !== -1)tinyUrlObjLocal.splice(index, 1);
+        }
+        if (changes.itemsAdded.length) {
+            if (tinyUrlObjLocal)tinyUrlObjLocal = [...tinyUrlObjLocal, ...changes.itemsAdded];
+            else tinyUrlObjLocal = changes.itemsAdded;
+        }
+        data.tinyUrlArray = tinyUrlObjLocal;
+        data.templateName = props.templateName ? props.templateName : "";
+        data.templateVersion = props.templateVersion ? props.templateVersion : "";
+        axios({
+            method: 'put',
```

Comment:
Capitalize `PUT`.
Reference: https://stackoverflow.com/a/27427356

Reply:
I used it from here:
https://www.npmjs.com/package/axios#axios-api